src/decompose.js

```diff
@@ -0,0 +1,66 @@
+import { scale } from './scale'
+import { compose } from './transform'
+/**
+ * Decompose a matrix into translation, rotation and scaling components, optionally
+ * take horizontal and vertical flip in to consideration.
+ * Note this function decomposes a matrix in rotation -> scale -> translation order. I.e. for
+ * certain translation T {tx, ty}, rotation R and scaling S { sx, sy }, it's only true for:
+ *  decompose(compose(T, S, R)) === { translate: T, rotation: R, scale: S }
```

Comment:
Since this function decomposes a subset of transformation matrices, I was wondering to rename it to `decomposeTSR()`. What do you think about?

Reply:
No problem.